autosklearn/evaluation/splitter.py

```diff
@@ -56,16 +57,43 @@ def _iter_indices(self, X, y, groups=None):  # type: ignore
             train = []
             test = []
 
+            # NOTE: Adapting for unique instances
+            #
+            #   Each list n_i, t_i represent the list of class in the
+            #   training_set and test_set resepectively.
+            #
+            #   n_i = [100, 100, 0, 3]  # 100 instance of class '0', 0 instance of class '2'
+            #   t_i = [300, 300, 1, 3]  # 300 instances of class '0', 1 instance of class '2'
+            #
+            #  To support unique labels such as class '2', which only has one sample
+            #  between both n_i and t_i, we need to make sure that n_i has at least
+            #  one sample of all classes. There is also the extra check to ensure
+            #  that the sizes stay the same.
+            #
+            #   n_i = [ 99, 100, 1, 3]  # 100 instance of class '0', 0 instance of class '2'
+            #            |       ^
+            #            v       |
+            #   t_i = [301, 300, 0, 3]  # 300 instances of class '0', 1 instance of class '2'
+            #
+            for i, class_count in enumerate(n_i):
+                if class_count == 0:
+                    t_i[i] -= 1
+                    n_i[i] += 1
+
+                    j = np.argmax(n_i)
+                    if n_i[j] == 1:
+                        warnings.warn("Can't respect size requirements for split"
```

Comment:
This only comes up in extreme cases, as we can read `n_i[j] == 1` as saying "the class j with the most instances in the training set, only has 1 instance". 

The reason we need one of each class in our training set is so that algorithms will always produce the correct output shape.

Here's an example where we would encounter this issue.

```python
# Initial with classes 'a', 'b', 'c', 
n_i = [2, 0, 0] # training set with 2 'a', 0 'b' and 0 'c'
t_i = [1, 1, 1] # test set with 1 of each

# We  have no instances of 'b' so we take one from the test set
n_i = [1, 1, 0]
       |  ^
       v  |
t_i = [2, 0, 1]

# We have no instances of class 'c' in the training set so we take one from the test set
# warning raised as we now break size requirements.
n_i = [1, 1, 1]
             ^
             |
t_i = [2, 0, 0]  
```

Here is where that warning would come up, we can't remove an instance from the training set without missing a class in our training set.

Reply:
Yes, I understand. But I cannot connect the warning message to the behavior. Why must it contain `any unique labels`?

Reply:
Perhaps it would be clearer as `The training set must contain all of the unique labels that exist in the dataset`?